model/task/db.go

```diff
@@ -674,6 +674,24 @@ func FindOneId(id string) (*Task, error) {
 	return task, nil
 }
 
+func FindOneIdAndExecution(id string, execution int) (*Task, error) {
```

Comment:
I think you can use FindOneIdOldOrNew instead

Reply:
I decided not to do this because we almost always want the task from the tasks collection. The bug this addresses happened 5 times out of 167629 tasks in the last 24 hours. If I use `FindOneIdOldOrNew` I will check the old tasks collection first, which is almost never what I want. Instead I want to fall back to the old tasks collection only if the tasks collection doesn't have what I want.

Reply:
Can you add a parameter to that function instead of writing a new one?

Reply:
maybe the right answer is to write a FindOneIdNewOrOld()